plugins/outputs/cloudwatchlogs/internal/pusher/target.go

```diff
@@ -109,26 +133,89 @@ func (m *targetManager) createLogStream(t Target) error {
 	return err
 }
 
-// PutRetentionPolicy tries to set the retention policy for a log group. Does not retry on failure.
-func (m *targetManager) PutRetentionPolicy(t Target) {
-	if t.Retention > 0 {
-		i := aws.Int64(int64(t.Retention))
-		putRetentionInput := &cloudwatchlogs.PutRetentionPolicyInput{
-			LogGroupName:    &t.Group,
-			RetentionInDays: i,
-		}
-		_, err := m.service.PutRetentionPolicy(putRetentionInput)
+func (m *targetManager) processRetentionUpdates() {
+	for target := range m.retentionPolicyUpdates {
+		err := m.processWithRetry(target)
 		if err != nil {
-			// since this gets called both before we start pushing logs, and after we first attempt
-			// to push a log to a non-existent log group, we don't want to dirty the log with an error
-			// if the error is that the log group doesn't exist (yet).
-			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
-				m.logger.Debugf("Log group %v not created yet: %v", t.Group, err)
-			} else {
-				m.logger.Errorf("Unable to put retention policy for log group %v: %v ", t.Group, err)
+			m.logger.Errorf("Failed to update retention policy for target %v: %v", target, err)
+		}
+	}
+}
+
+func (m *targetManager) processWithRetry(target Target) error {
+	var err error
+	for attempt := 0; attempt < maxAttempts; attempt++ {
+		err = m.checkAndUpdate(target)
+		if err == nil {
+			return nil
+		}
+
+		m.logger.Debugf("retrying to update retention policy for target (%v) %v: %v", attempt, target, err)
+		delay := m.calculateBackoff(attempt)
+		time.Sleep(delay)
+	}
+	return fmt.Errorf("failed to update retention policy after %d attempts: %w", maxAttempts, err)
+}
+
+func (m *targetManager) checkAndUpdate(target Target) error {
+	currentRetention, err := m.describeLogGroupRetention(target)
```

Comment:
Can likely skip this API call when it's a new log group.
https://github.com/aws/amazon-cloudwatch-agent/blob/799f0c3bdcfc18fcadf3b5e0c1c9ebaa543e18a8/plugins/outputs/cloudwatchlogs/internal/pusher/target.go#L43-L55